tests/pos/i3471.scala

```diff
@@ -1 +0,0 @@
-sealed object Fun
```

Comment:
This testcase was added in #3486 for #3471 to ensure the compiler didn't crash.
During the discussion, it was demanded that `final` be handled at the same time: https://github.com/lampepfl/dotty/pull/3486#discussion_r151471447.